tests/test_misc.py

```diff
@@ -1145,6 +1145,57 @@ def test_rescan(node_factory, bitcoind):
     assert not l1.daemon.is_in_log(r'Adding block 102')
 
 
+@pytest.mark.xfail(strict=True)
+def test_bitcoind_goes_backwards(node_factory, bitcoind):
+    """Check that we refuse to acknowledge bitcoind giving a shorter chain without explicit rescan"""
+    l1 = node_factory.get_node(may_fail=True, allow_broken_log=True)
+
+    bitcoind.generate_block(10)
+    sync_blockheight(bitcoind, [l1])
+    l1.stop()
+
+    # Now shrink chain (invalidateblock leaves 'headers' field)
```

Comment:
Just tried it myself, and indeed the `headers` remains unchanged after `invalidateblock`, however if `invalidateblock` is followed by a restart `headers` will reset to the expected value. Maybe that's a more stable way of going backwards than blowing away the underlying directory?

Reply:
Ah, that would have been easier!